src/viam/errors.py

```diff
@@ -83,3 +83,13 @@ class NotSupportedError(ViamGRPCError):
     def __init__(self, message: str):
         self.message = message
         self.grpc_code = Status.UNIMPLEMENTED
+
+
+class ValidationError(ViamGRPCError):
+    """
+    Exception raised when there is an error during module validation
+    """
+
+    def __init__(self, message: str):
+        self.message = message
+        self.grpc_code = Status.FAILED_PRECONDITION
```

Comment:
I'm not entirely sure which Status to use here.

Reply:
`FAILED_PRECONDITION` seems fine to me. Go and C++ both return `UNKNOWN`.

Reply:
I think `FAILED_PRECONDITION` is actually the wrong status here https://grpc.github.io/grpc/core/md_doc_statuscodes.html. The system itself is not in a bad state, so maybe `INVALID_ARGUMENT`?

Reply:
Agree, I think `INVALID_ARGUMENT` is better